report/3.Theory/ObjectRecognition.tex

```diff
@@ -0,0 +1,133 @@
+The first layer could recognize lines, edges or similar low level features.
+The following layers could then detect increasingly complicated facial features, ending at some layer, in which one neuron would transmit a signal if a nose is present in the image, and another neuron to transmit if a mouth is present.
+
+
+\figur{0.6}{images/cnn_face}{A simplified example of the abstractions within a CNN
+	\cite{CNNStructure}.}{fig:face_cnn}
+
+However, it is important to understand, that this is not necessarily the patterns that a CNN will detect within its layers.
+Actually it is far more likely, that the layers will recognize something that to a human has no relation to a face.
+But the computer will have found a correlation between these seemingly random patterns, and a face.
+
+CNNs are often used along with pooling \cite{CNN}.
+Pooling is a concept in which multiple outputs are combined into a single neuron in the next layer.
+This can be done in order to down-sample the input and reduce the complexity making the CNN run faster.
+Some common examples of pooling is max pooling and average pooling, in which the average or max value is output from a number of input neurons.
+
+CNNs are an effective choice for classification. 
+This means it can answer yes or no to whether the image is of a car.
+Now the question is whether it can be used to figure out the position of the objects within the image.
+The answer turns out to be yes.
+
+
+\subsubsection{Regional Convolutional Neural Network}
+
+A regional convolutional neural network (RCNN) utilizes CNN to get both a classification along with a bounding box, which in turn can be used to determine the center of the location.
+It does this by applying the CNN to smaller regions of the image. 
+These regions are extracted from the image using a method called selective search.
+If the region contains an element recognized by the CNN a bounding box can be drawn around the region.
+The bounding box can at last be tightened using different methods, a popular option being a linear regression model.\cite{CNNHistory}
+
+Unfortunately RCNN in itself is not viable for real time object detection.
+However, there have been a lot of modifications and improvements of RCNN that improves the performance of it.
+One improvement is called fast RCNN and is leaning toward being feasible for real time object detection.
+
+%\subsection{You Only Look Once}
+
+%An alternative to RCNN is You Only Look Once (YOLO) which is a new technology developed with real time processing in mind.
+
+
+
+\subsection{GOTURN}
+% https://www.learnopencv.com/goturn-deep-learning-based-object-tracking/
+Generic Object Tracking Using Regression Networks, is an algorithm based on deep learning\cite{goturn}.
+GOTURN is different than the majority of algorithms presented in this section, in that it uses offline learning, whereas online learning constantly improves the model during run-time.
```

Comment:
have we talked about offline learning?

Reply:
maybe it is more that online learning isn't talked about. also it isnt clear that the below describes offline learning